### PR TITLE
Remove Dispose and Activate on ObjectAdapter

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR removes these two sync "helper" methods. You should only call (and await) the Async methods.

### CODE REVIEW

File: csharp/test/Ice/binding/RemoteObjectAdapterI.cs [hunk 1]
```diff
@@ -14,20 +15,12 @@ public RemoteObjectAdapter(ObjectAdapter adapter)
...
-            try
-            {
-                _adapter.Dispose();
-            }
-            catch (ObjectDisposedException)
-            {
-            }
-        }
+        public async ValueTask DeactivateAsync(Current current, CancellationToken cancel) =>
+            await _adapter.DisposeAsync();
```
Comment (bentoi) [inline csharp/test/Ice/binding/RemoteObjectAdapterI.cs:24]: Do we really need the async/await here? can't we just return the value task from `_adapter.DisposeAsync()`?

**Review by bentoi — APPROVED** [review]
Looks good!

**Review by pepone — APPROVED** [review]
Looks good 👍

File: csharp/test/Ice/binding/RemoteObjectAdapterI.cs (hunk 1 above)
Comment (bernardnormier) [reply csharp/test/Ice/binding/RemoteObjectAdapterI.cs:24]: Fixed, we don't need it.

**Pull request merged**